idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2296 @@
+    ")\n",
+    "\n",
+    "# Import IDAES cores\n",
+    "from idaes.core import (\n",
+    "    declare_process_block_class,\n",
+    "    MaterialFlowBasis,\n",
+    "    PhysicalParameterBlock,\n",
+    "    StateBlockData,\n",
+    "    StateBlock,\n",
+    "    MaterialBalanceType,\n",
+    "    EnergyBalanceType,\n",
+    "    Solute,\n",
+    "    Solvent,\n",
+    "    LiquidPhase,\n",
+    ")\n",
+    "\n",
+    "# Some more information about this module\n",
+    "__author__ = \"Javal Vyas\"\n",
+    "\n",
+    "\n",
+    "# Set up logger\n",
+    "_log = logging.getLogger(__name__)\n",
+    "\n",
+    "\n",
+    "@declare_process_block_class(\"AqPhase\")\n",
+    "class PhysicalParameterData(PhysicalParameterBlock):\n",
+    "    \"\"\"\n",
+    "    Property Parameter Block Class\n",
+    "\n",
+    "    Contains parameters and indexing sets associated with properties for\n",
+    "    liquid Phase\n",
+    "\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def build(self):\n",
+    "        \"\"\"\n",
+    "        Callable method for Block construction.\n",
+    "        \"\"\"\n",
+    "        super().build()\n",
+    "\n",
+    "        self._state_block_class = AqPhaseStateBlock\n",
+    "\n",
+    "        # List of valid phases in property package\n",
+    "        self.Aq = LiquidPhase()\n",
+    "\n",
+    "        # Component list - a list of component identifiers\n",
+    "        self.NaCl = Solute()\n",
+    "        self.KNO3 = Solute()\n",
+    "        self.CaSO4 = Solute()\n",
+    "        self.H2O = Solvent()\n",
+    "\n",
+    "        self.solutes = Set(initialize=[\"NaCl\", \"KNO3\", \"CaSO4\"])\n",
+    "        # Heat capacity of solvent\n",
+    "        self.cp_mass = Param(\n",
+    "            mutable=True,\n",
+    "            initialize=4182,\n",
+    "            doc=\"Specific heat capacity of solvent\",\n",
+    "            units=units.J / units.kg / units.K,\n",
+    "        )\n",
+    "\n",
+    "        self.dens_mass = Param(\n",
+    "            mutable=True,\n",
+    "            initialize=997,\n",
+    "            doc=\"Density of ethylene dibromide\",\n",
+    "            units=units.kg / units.m**3,\n",
+    "        )\n",
+    "        self.temperature_ref = Param(\n",
+    "            within=PositiveReals,\n",
+    "            mutable=True,\n",
+    "            default=298.15,\n",
+    "            doc=\"Reference temperature\",\n",
+    "            units=units.K,\n",
+    "        )\n",
+    "\n",
+    "    @classmethod\n",
+    "    def define_metadata(cls, obj):\n",
+    "        obj.add_default_units(\n",
+    "            {\n",
+    "                \"time\": units.hour,\n",
+    "                \"length\": units.m,\n",
+    "                \"mass\": units.g,\n",
+    "                \"amount\": units.mol,\n",
+    "                \"temperature\": units.K,\n",
+    "            }\n",
+    "        )\n",
+    "\n",
+    "\n",
+    "class _StateBlock(StateBlock):\n",
```

Comment:
I note that here you are defining `fix_initialization_states` whilst above you defined `release_state` (for the other property package). I think you need to be consistent about what methods are needed.